File: gherkin/java/src/main/java/gherkin/ast/StepsContainer.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public abstract class StepsContainer extends Node {
    private final String keyword;
    private final String name;
    private final String description;
    private final List<Step> steps;

    StepsContainer(Location location, String keyword, String name, String description, List<Step> steps) {
        super(location);
        this.keyword = keyword;
        this.name = name;
        this.description = description;
        this.steps = Collections.unmodifiableList(steps);
    }

    public String getName() {
        return name;
    }

    public String getKeyword() {
        return keyword;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public List<Step> getChildren() {
        return steps;
    }
}

File: gherkin/dotnet/Gherkin/Events/Args/Ast/StepsContainer.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using Gherkin.CucumberMessages.Types;

namespace Gherkin.Events.Args.Ast
{
    public class StepsContainer
    {
        public StepsContainer()
        {
            Steps = new List<Step>();
            Id = IdGenerator.GetNextId();
        }

        [DataMember(Name = "id")]
        public string Id { get; set; }

        [DataMember(Name = "keyword")]
        public string Keyword { get; set; }
        [DataMember(Name = "location")]
        public CucumberMessages.Types.Location Location { get; set; }
        [DataMember(Name = "name")]
        public string Name { get; set; }
        [DataMember(Name = "description")]
        public string Description { get; set; }
        [DataMember(Name = "steps")]
        public IReadOnlyCollection<Step> Steps { get; set; }

        [DataMember(Name = "examples")]
        public IReadOnlyCollection<Examples> Examples { get; set; }

        [DataMember(Name = "tags")]
        public IReadOnlyCollection<Tag> Tags { get; set; }
    }
}